### PR TITLE
yaml file: bonding tests with extended bonding module parameter options.

### PR DESCRIPTION
**Pull request by vaishnavibhat**
yaml file to include bongding module parameters for extended tests.

Signed-off-by: Vaishnavi Bhat <vaishnavi@linux.vnet.ibm.com>

### CODE REVIEW

File: io/net/bonding.py [hunk 1]
```diff
@@ -80,6 +80,24 @@ def setUp(self):
         self.peer_first_ipinterface = self.params.get("peer_ip", default="")
         if not self.peer_interfaces or self.peer_first_ipinterface == "":
             self.cancel("peer machine should available")
+
+        self.xmit_hash_policy = self.params.get("xmit_hash_policy",
+                                                default="1")
+        self.miimon = self.params.get("miimon", default="")
```
Comment (narasimhan-v) [inline io/net/bonding.py]: default please.

File: io/net/bonding.py [hunk 2]
```diff
@@ -80,6 +80,24 @@ def setUp(self):
...
+        self.fail_over_mac = self.params.get("fail_over_mac",
+                                             default="2")
+        self.downdelay = self.params.get("downdelay", default="0")
+        self.packets_per_slave = self.params.get("packets_per_slave",
+                                                 default="2")
+        self.resend_igmp = self.params.get("resend_igmp", default="1")
+        self.tlb_dynamic_lb = self.params.get("tlb_dynamic_lb",
+                                              default="1")
+        self.primary = self.params.get("primary", default="")
```
Comment (narasimhan-v) [inline io/net/bonding.py]: default here also

File: io/net/bonding.py [hunk 3]
```diff
@@ -80,6 +80,24 @@ def setUp(self):
...
+        self.primary_reselect = self.params.get("primary_reselect",
+                                                default="0")
+        self.num_unsol_na = self.params.get("num_unsol_na", default="1")
+        self.lp_interval = self.params.get("lp_interval", default="1")
+        self.primary = self.params.get("primary", default="")
```
Comment (narasimhan-v) [inline io/net/bonding.py]: is this a repeat ?

File: io/net/bonding.py [hunk 4]
```diff
@@ -239,8 +257,34 @@ def bond_setup(self, arg1, arg2):
...
+            if self.mode == '1' or self.mode == '5' or self.mode == '6':
+                genio.write_file("%s/bonding/primary" % self.bond_dir, self.primary)
+                genio.write_file("%s/bonding/primary_reselect" % self.bond_dir,
+                                 self.primary_reselect)
+            if self.mode == '0' or self.mode == '1' or self.mode == '5' or self.mode == '6':
+                genio.write_file("%s/bonding/resend_igmp" % self.bond_dir, self.resend_igmp)
+            if self.mode == '2' or self.mode == '4' or self.mode == '5':
+                genio.write_file("%s/bonding/xmit_hash_policy" % self.bond_dir, self.xmit_hash_policy)
+            if self.mode == '5' or self.mode == '6':
+                genio.write_file("%s/bonding/lp_interval" % self.bond_dir, self.lp_interval)
```
Comment (narasimhan-v) [inline io/net/bonding.py]: @vaishnavibhat Can this handling be done in a more robust way ?

**Comment by narasimhan-v** [comment]
@vaishnavibhat can you check if the latest code has been pushed ?

**Comment by PraveenPenguin** [comment]
@vaishnavibhat this PR has conflict with the latest code if you still persuing this PR please raise new one

**Pull request closed without merging**